pkg/status/controller_test.go

```diff
@@ -0,0 +1,346 @@
+			existingCO: &configv1.ClusterOperator{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: clusterOperatorName,
+				},
+				Status: configv1.ClusterOperatorStatus{
+					Conditions: []configv1.ClusterOperatorStatusCondition{
+						{
+							Type:               configv1.OperatorAvailable,
+							Status:             configv1.ConditionFalse,
+							LastTransitionTime: metav1.Now(),
+							Reason:             "",
+							Message:            "",
+						},
+						{
+							Type:               configv1.OperatorDegraded,
+							Status:             configv1.ConditionTrue,
+							LastTransitionTime: metav1.Now(),
+							Reason:             "",
+							Message:            "",
+						},
+						{
+							Type:               configv1.OperatorProgressing,
+							Status:             configv1.ConditionTrue,
+							LastTransitionTime: metav1.Now(),
+							Reason:             "",
+							Message:            "",
+						},
+						{
+							Type:               configv1.OperatorUpgradeable,
+							Status:             configv1.ConditionFalse,
+							LastTransitionTime: metav1.Now(),
+							Reason:             "",
+							Message:            "",
+						},
+					},
+					Versions: []configv1.OperandVersion{
+						{
+							Name:    "anything",
+							Version: "anything",
+						},
+					},
+					RelatedObjects: []configv1.ObjectReference{
+						{
+							Group:    "",
+							Resource: "anything",
+							Name:     "anything",
+						},
+					},
+				},
+			},
+		}),
+	)
+})
+
+var _ = Describe("Filters are exclusive", func() {
+	It("Should allow only 'machine-approver' clusterOperators", func() {
+		operator := &configv1.ClusterOperator{}
+		operator.SetName(clusterOperatorName)
+		Expect(clusterOperatorFilter(operator)).To(BeTrue())
+		operator.SetName("unknown")
+		Expect(clusterOperatorFilter(operator)).To(BeFalse())
+		Expect(clusterOperatorFilter(nil)).To(BeFalse())
+	})
+
+	It("Should allow only CSR objects", func() {
+		Expect(csrFilter(&certificatesv1.CertificateSigningRequest{})).To(BeTrue())
+		Expect(csrFilter(nil)).To(BeFalse())
+	})
+})
+
+var _ = Describe("toClusterOperator mapping is targeting requests to 'machine-approver' clusterOperator", func() {
+	It("Should map reconciles to 'machine-approver' CO", func() {
+		object := certificatesv1.CertificateSigningRequest{}
+		mapObject := handler.MapObject{
+			Meta:   object.GetObjectMeta(),
+			Object: &object,
+		}
+		requests := []reconcile.Request{{
+			NamespacedName: client.ObjectKey{
+				Name: clusterOperatorName,
+			},
+		}}
+		Expect(toClusterOperator(mapObject)).To(Equal(requests))
+	})
+})
+
+var _ = Describe("Reconcile triggers", func() {
+	var operator *configv1.ClusterOperator
+	var csr *certificatesv1.CertificateSigningRequest
+	var env *envtest.Environment
+	var c client.Client
+	var done chan struct{}
+	var reconciled int
+	var reconciledTimes = func() int { return reconciled }
+
+	BeforeEach(func() {
+		reconciled = 0
```

Comment:
Similar concerns to the other package tests, perhaps this would be better as a channel?